doc/qos/ECN_and_WRED_statistics_HLD.md

```diff
@@ -0,0 +1,454 @@
+<p align=center>
+<img src="ecn-wred-stats-images/port_counter_cli_flow.png" alt="port CLI interactions">
+</p>
+
+
+
+### SAI API
+
+Following SAI statistics are used in this feature,
+
+* SAI counters list,
+    * SAI_QUEUE_STAT_WRED_ECN_MARKED_PACKETS
+    * SAI_QUEUE_STAT_WRED_ECN_MARKED_BYTES
+    * SAI_QUEUE_STAT_WRED_DROPPED_PACKETS
+    * SAI_QUEUE_STAT_WRED_DROPPED_BYTES
+    * SAI_PORT_STAT_GREEN_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_YELLOW_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_RED_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_WRED_DROPPED_PACKETS
+    * SAI_PORT_STAT_ECN_MARKED_PACKETS [to be supported in next phase of Enhancement]
+
+* sai_query_stats_capability() will be used to identify the supported statistics
+
+* For statistics get and clear, existing APIs are used as it is
+
+### Configuration and management
+
+The config CLI commands ```"counterpoll wredqueue <enable | disable>"``` and ```"counterpoll wredport <enable | disable>"``` will enable and disable the WRED and ECN statistics by interacting with the FLEX_COUNTER_TABLE table of CONFIG_DB.
+
+### Manifest (if the feature is an Application Extension)
+Not applicable?
+
+### CLI/YANG model Enhancements
+The sonic-flex_counter.yang will be updated with new containers to reflect the proposed CONFIG_DB changes as shown below,
+
+        {
+            container WRED_ECN_QUEUE {
+                /* WRED_ECN_QUEUE_FLEX_COUNTER_GROUP */
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+                leaf FLEX_COUNTER_DELAY_STATUS {
+                    type flex_delay_status;
+                }
+                leaf POLL_INTERVAL {
+                    type poll_interval;
+                }
+            },
+            container WRED_ECN_PORT {
+                /* WRED_ECN_PORT_FLEX_COUNTER_GROUP */
+                leaf FLEX_COUNTER_STATUS {
+                    type flex_status;
+                }
+                leaf FLEX_COUNTER_DELAY_STATUS {
+                    type flex_delay_status;
+                }
+                leaf POLL_INTERVAL {
+                    type poll_interval;
+                }
+            }
+        }
+
+
+
+### Warmboot and Fastboot Design Impact
+There are no impact to warmboot or fastboot.
+
+
+### Restrictions/Limitations
+
+* None
+
+### Testing Requirements/Design
+
+#### Unit Test cases
+- On Supported platforms, Verify if the queuestat CLI display has WRED and ECN Queue statistics
+- On Supported platforms, Verify if the port statistcs CLI display has the WRED statistics
+- On ECN-WRED stats non-supported platforms,
+    - Verify that CLI does not show the respective Headers in queue stats
+    - Verify that CLI does not show the respective rows in port stats
+    - Verify that the stat capability get is not logging errors to syslog
+
+#### System Test cases
+* New sonic-mgmt(PTF) ecn wred statistics testcase will be created to verify the statistics on supported platforms

```

Comment:
@rpmarvell could you please also share the sonic-mgmt PR?

Reply:
URL for sonic-mgmt PR : https://github.com/sonic-net/sonic-mgmt/pull/8179